src/main/java/org/jitsi/jigasi/TranscriptionGateway.java

```diff
@@ -92,35 +140,51 @@ public void stop()
     }
 
     /**
-     * Tries to retrieve a transcriber assigned to a tenant
-     * if the property value is a json. Returns the value as
-     * is if no JSON is found.
+     * Tries to retrieve a custom defined transcriber by checking
+     * multiple sources. First it issues a GET request to the
+     * remoteTranscriptionConfigUrl property if defined. It expects
+     * a JSON response with a transcriberType property. If not found
+     * it tries to read the transcription.customService property. If
+     * that also fails it returns the default GoogleCloudTranscriptionService.
      *
      * @param tenant the tenant which is retrieved from the context
+     * @param roomJid the roomJid which is retrieved from the context (used only with JaaS)
      */
-    private String getCustomTranscriptionServiceClass(String tenant)
+    private String getCustomTranscriptionServiceClass(String tenant, String roomJid)
     {
         String transcriberClass = null;
-        String remoteTranscriptionConfigUrl
-                = JigasiBundleActivator.getConfigurationService()
-                .getString(
-                        REMOTE_TRANSCRIPTION_CONFIG_URL,
-                        null);
 
-        if (remoteTranscriptionConfigUrl != null && tenant != null)
+        if (remoteTranscriptionConfigUrl != null)
         {
-            String tsConfigUrl = remoteTranscriptionConfigUrl + "/" + tenant;
+            String tsConfigUrl;
+
+            // this is JaaS specific
+            if (remoteTranscriptionConfigUrl.contains("jitsi.net"))
```

Comment:
Why do we need that, can't we use the same format of the URL always?

Reply:
Not really. As you some of the clients will have the ability to choose a specific transcriber on a per room basis.

Reply:
Can we always do /tenanat?roomname= ... Hardcoding dns here is bot something we want.